run/retry.go

```diff
@@ -0,0 +1,94 @@
+package run
+
+import (
+	"context"
+	"math/rand"
+	"time"
+)
+
+// WithRetry enables an application to handle transient failures by transparently retrying a failed operation.
+func WithRetry(backoff []time.Duration, classifier func(error) Result, fn func(context.Context) error) func(context.Context) error {
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	return func(ctx context.Context) error {
+		retries := 0
+
+		for {
+			err := fn(ctx)
+			retries++
```

Comment:
backoff[0] will never be used. It works because the number of sleeps equals the number of attempts minus 1

Reply:
Improved the implementation, hope now it's more clear.